packages/@aws-cdk/lambda-layer-kubectl/lib/kubectl-layer.ts

```diff
@@ -10,18 +9,11 @@ import { Construct } from 'constructs';
 export class KubectlLayer extends lambda.LayerVersion {
   constructor(scope: Construct, id: string) {
     super(scope, id, {
-      code: lambda.Code.fromAsset(path.join(__dirname, 'layer.zip'), {
-        // we hash the Dockerfile (it contains the tools versions) because hashing the zip is non-deterministic
-        assetHash: hashFile(path.join(__dirname, '..', 'layer', 'Dockerfile')),
+      code: lambda.Code.fromAsset(ASSET_FILE, {
+        // we hash the layer directory (it contains the tools versions and Dockerfile) because hashing the zip is non-deterministic
+        assetHash: FileSystem.fingerprint(LAMBDA_SOURCE_DIR),
```

Comment:
@rix0rrr how do you feel about changing the way we are hashing to be in line with what we have in the awscli layer: https://github.com/aws/aws-cdk/blob/main/packages/@aws-cdk/lambda-layer-awscli/lib/awscli-layer.ts#L14.

This would obviously change the asset hash when users update their aws-cdk-lib.

Reply:
I tested this independently, which shows that a lambda layer with an updated hash and no additional changes will successfully deploy, replacing the previous version.